### PR TITLE
Skeletons for showing statistics in the home screen feed (EXPOSUREAPP-3674)

### PR DESCRIPTION
**Pull request by d4rken**
Introduce base for showing statistics in the home screen feed to enable parallel work on each screen.
Adds a nested horizontally scrolling RecyclerView that supports dynamic content updates.

### Testing
* It should not crash
* The code should not have any broad flaws that will be a problem later on

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/source/StatisticsProvider.kt [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+@Singleton
+class StatisticsProvider @Inject constructor() {
+
+    private val currentInternal = MutableStateFlow(StatisticsData(items = emptyList()))
+    val current: Flow<StatisticsData> = currentInternal.filterNotNull()
+
+    init {
+        // Mock data
+        GlobalScope.launch(context = Dispatchers.IO) {
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/source/StatisticsProvider.kt:26]: it is better to inject the dispatcher

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/ui/homecards/StatisticsCardPaddingDecorator.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                if (itemPosition == it.itemCount - 1) {
+                    outRect.right = resources.getDimensionPixelSize(endPadding)
+                } else {
+                    val distance = resources.getDimensionPixelSize(cardDistance)
+                    outRect.left = distance / 2
+                    outRect.right = distance / 2
+                }
+            }
+        }
+    }
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/ui/homecards/StatisticsCardPaddingDecorator.kt:37]: maybe:
```
when(itemPosition){
}
``` 
is more readable

File: Corona-Warn-App/src/main/res/layout/home_statistics_cards_basecard_layout.xml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.cardview.widget.CardView xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/home_statistics_cards_basecard_layout.xml:2]: [Optional] there is MaterailCardView , something to keep in mind towards MDC

File: Corona-Warn-App/src/main/res/layout/home_statistics_cards_incidence_layout.xml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+            android:layout_height="wrap_content"
+            android:layout_marginStart="24dp"
+            android:layout_marginTop="24dp"
+            android:focusable="false"
+            android:text="Incidence"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
+    </androidx.constraintlayout.widget.ConstraintLayout>
+
+</layout>
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/home_statistics_cards_incidence_layout.xml:24]: is layout tag needed here?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/source/StatisticsProvider.kt (hunk 1 above)
Comment (d4rken) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/source/StatisticsProvider.kt:26]: Agree, but this is all just placeholders so multiple people can start working on this, I'll rewrite the whole class next week.

File: Corona-Warn-App/src/main/res/layout/home_statistics_cards_basecard_layout.xml (hunk 3 above)
Comment (d4rken) [reply Corona-Warn-App/src/main/res/layout/home_statistics_cards_basecard_layout.xml:2]: I think for that we also need the MaterialTheme (using that for the diary), not AppCompatTheme (rest of the app).
Can we use the MaterialCardView without that?
We have a ticket for refactoring the theming open.

File: Corona-Warn-App/src/main/res/layout/home_statistics_cards_incidence_layout.xml (hunk 4 above)
Comment (d4rken) [reply Corona-Warn-App/src/main/res/layout/home_statistics_cards_incidence_layout.xml:24]: In the current form, no, but it's a placeholder. The final layout probably needs it, but the person that will work on the ticket for this card will have to decide that.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/ui/homecards/StatisticsCardPaddingDecorator.kt (hunk 2 above)
Comment (d4rken) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/ui/homecards/StatisticsCardPaddingDecorator.kt:37]: Not sure due to the `adapter.?` but will try. :ok_hand:

Comment (d4rken) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/ui/homecards/StatisticsCardPaddingDecorator.kt:38]: Done.

File: Corona-Warn-App/src/main/res/layout/home_statistics_cards_basecard_layout.xml (hunk 3 above)
Comment (mtwalli) [reply Corona-Warn-App/src/main/res/layout/home_statistics_cards_basecard_layout.xml:2]: Not sure, let us keep for the refactoring ticket then

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/source/StatisticsProvider.kt (hunk 1 above)
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/source/StatisticsProvider.kt:26]: ok

File: Corona-Warn-App/src/main/res/layout/home_statistics_cards_incidence_layout.xml (hunk 4 above)
Comment (mtwalli) [reply Corona-Warn-App/src/main/res/layout/home_statistics_cards_incidence_layout.xml:24]: make sense

**Review by mtwalli — APPROVED** [review]
👍🏼

**Review by ralfgehrer — APPROVED** [review]

**Pull request merged**